tests/constraints/test_decimal_constraints.py

```diff
@@ -52,7 +52,7 @@ def test_handle_constrained_decimal_length_validation() -> None:
 def test_handle_constrained_decimal_handles_max_digits(max_digits: int) -> None:
     if max_digits > 0:
         result = handle_constrained_decimal(create_constrained_field(max_digits=max_digits))
-        assert len(result.as_tuple().digits) - abs(result.as_tuple().exponent) <= max_digits
+        assert len(result.as_tuple().digits) - abs(cast("int", result.as_tuple().exponent)) <= max_digits  # type: ignore[redundant-cast]
```

Comment:
I am using `cast` here since it should be impossible to generate a decimal with string exponent in `handle_constrained_decimal` so even if it happens it means we have a bug and the test will fail anyway because `abs` will throw an exception in this case. 

`# type: ignore[redundant-cast]` is needed for mypy because it contains an old version of typeshed which still types `result.as_tuple().exponent` as `int` while pyright bundles more recent version. Once new mypy is released we can remove this ignore comment.